codes/quantum/oscillators/chuang-leung-yamamoto.yml

```diff
@@ -0,0 +1,59 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+
+# code id, physical, logical are all lower case
+code_id: chuang-leung-yamamoto
+physical: oscillators
+logical: qubits
+
+name: 'Chuang-Leung-Yamamoto'
+
+# First reference with the code (optional).
+introduced: '\cite{doi:10.1103/PhysRevA.56.1114}'
+
+# Anything applicable to a larger parent set of codes (see below) should go in
+# that entry instead of here.
+description: 'Codes parametrized by \([[N,M,2^k,d]]\). Encodes \(k\) qubits with distance \(d\) in \(m\) bosonic modes using states with total excitation number \(N\). A given Fock state will appear in at most one codeword.'
```

Comment:
Bosonic qubit-into-oscillator codes that encode k qubits into M oscillators, with each oscillator restricted to having at most N excitations. Code states are expressed as superpositions of oscillator Fock states, and a code distance can be defined as ...  (note that new entry fock_state muses on the distance as well). Codes can be denoted as [[N,M,2^k,d]], which conflicts with \hyperref[code:stabilizer]{stabilizer code} notation.